conn.go

```diff
@@ -166,9 +166,15 @@ func (s *Swarm) addConn(netConn net.Conn, isServer bool) (*Conn, error) {
 	// unlocked **before** the Handler is called (which needs to be
 	// sequential). This was the simplest thing :)
 	setupConn := func() (*Conn, error) {
+
+		// create a new spdystream connection
+		ssConn, err := s.transport.NewConn(netConn, isServer)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
why is this getting moved up?? i dont think it should. see the note below:

```
// first, check if we already have it...
```

`transport.NewConn` could be an expensive operation.